server/core/include/irods_delay_queue.hpp

```diff
@@ -1,47 +1,78 @@
 #ifndef IRODS_DELAY_QUEUE_HPP
 #define IRODS_DELAY_QUEUE_HPP
 
+#include "fixed_buffer_resource.hpp"
+
+#include "boost/container/pmr/unsynchronized_pool_resource.hpp"
+#include "boost/container/pmr/global_resource.hpp"
+#include "boost/container/pmr/string.hpp"
+#include "boost/container/pmr/vector.hpp"
+
+#include <cstddef>
 #include <algorithm>
 #include <mutex>
-#include <string>
 #include <vector>
+#include <memory>
 
 namespace irods
 {
     class delay_queue
     {
     public:
-        delay_queue() = default;
+        explicit delay_queue(std::int64_t _pool_size_in_bytes)
+            : rules_mutex_{}
+            , buffer_{}
+            , upstream_allocator_{}
+            , allocator_{}
+            , queued_rules_{}
+        {
+            namespace bpmr = boost::container::pmr;
+            namespace ipmr = experimental::pmr;
+
+            if (_pool_size_in_bytes > 0) {
+                buffer_.resize(_pool_size_in_bytes);
+                upstream_allocator_.reset(new ipmr::fixed_buffer_resource(buffer_.data(), buffer_.size(), 250));
```

Comment:
what is the 250?

Reply:
that is the initial number of freelist pointers available to the memory resource. im not sure if admins should have access to it. the freelist pointers are allowed to allocate as much memory as possible to track allocations. at some point, the memory limit will kick end and enforce an indirect cap on the freelist pointers.

Reply:
right, no need to pull it out as a config.    maybe just a variable with a good name to convey the meaning.

Reply:
cool

Reply:
done